pkg/apis/contrail/v1alpha1/kubemanager_types.go

```diff
@@ -62,26 +62,27 @@ type KubemanagerServiceConfiguration struct {
-	ServiceAccount        string       `json:"serviceAccount,omitempty"`
-	ClusterRole           string       `json:"clusterRole,omitempty"`
-	ClusterRoleBinding    string       `json:"clusterRoleBinding,omitempty"`
-	CloudOrchestrator     string       `json:"cloudOrchestrator,omitempty"`
-	KubernetesAPIServer   string       `json:"kubernetesAPIServer,omitempty"`
-	KubernetesAPIPort     *int         `json:"kubernetesAPIPort,omitempty"`
-	KubernetesAPISSLPort  *int         `json:"kubernetesAPISSLPort,omitempty"`
-	PodSubnets            string       `json:"podSubnets,omitempty"`
-	ServiceSubnets        string       `json:"serviceSubnets,omitempty"`
-	KubernetesClusterName string       `json:"kubernetesClusterName,omitempty"`
-	IPFabricSubnets       string       `json:"ipFabricSubnets,omitempty"`
-	IPFabricForwarding    *bool        `json:"ipFabricForwarding,omitempty"`
-	IPFabricSnat          *bool        `json:"ipFabricSnat,omitempty"`
-	KubernetesTokenFile   string       `json:"kubernetesTokenFile,omitempty"`
-	HostNetworkService    *bool        `json:"hostNetworkService,omitempty"`
-	RabbitmqUser          string       `json:"rabbitmqUser,omitempty"`
-	RabbitmqPassword      string       `json:"rabbitmqPassword,omitempty"`
-	RabbitmqVhost         string       `json:"rabbitmqVhost,omitempty"`
+	Containers            []*Container       `json:"containers,omitempty"`
+	UseKubeadmConfig      *bool              `json:"useKubeadmConfig,omitempty"`
+	ServiceAccount        string             `json:"serviceAccount,omitempty"`
+	ClusterRole           string             `json:"clusterRole,omitempty"`
+	ClusterRoleBinding    string             `json:"clusterRoleBinding,omitempty"`
+	CloudOrchestrator     string             `json:"cloudOrchestrator,omitempty"`
+	KubernetesAPIServer   string             `json:"kubernetesAPIServer,omitempty"`
+	KubernetesAPIPort     *int               `json:"kubernetesAPIPort,omitempty"`
+	KubernetesAPISSLPort  *int               `json:"kubernetesAPISSLPort,omitempty"`
+	PodSubnets            string             `json:"podSubnets,omitempty"`
+	ServiceSubnets        string             `json:"serviceSubnets,omitempty"`
+	KubernetesClusterName string             `json:"kubernetesClusterName,omitempty"`
+	IPFabricSubnets       string             `json:"ipFabricSubnets,omitempty"`
+	IPFabricForwarding    *bool              `json:"ipFabricForwarding,omitempty"`
+	IPFabricSnat          *bool              `json:"ipFabricSnat,omitempty"`
+	KubernetesTokenFile   string             `json:"kubernetesTokenFile,omitempty"`
+	HostNetworkService    *bool              `json:"hostNetworkService,omitempty"`
+	RabbitmqUser          string             `json:"rabbitmqUser,omitempty"`
+	RabbitmqPassword      string             `json:"rabbitmqPassword,omitempty"`
+	RabbitmqVhost         string             `json:"rabbitmqVhost,omitempty"`
+	AuthMode              AuthenticationMode `json:"authMode,omitempty"`
```

Comment:
Is it possible to make in enum that only noauth and keystone values will be accepted?